jekyll/add-ons.markdown

```diff
@@ -324,7 +326,7 @@ module RubyLsp
     class Addon < ::RubyLsp::Addon
       def activate(global_state, message_queue)
         # Register the enhancement as part of the indexing process
-        @index.register_enhancement(MyIndexingEnhancement.new)
+        @index.register_enhancement(MyIndexingEnhancement.new(@index))
```

Comment:
I noticed when updating ruby-lsp-rails in https://github.com/Shopify/ruby-lsp-rails/pull/495 that it's a bit odd to pass `@index` to `@index`. What if we change this so that we just pass the class, i.e. `@index.register_enhancement(MyIndexingEnhancement)`, and let `register_enhancement` instantiate it?

Reply:
I thought about that but it means addon authors will lose the ability to customize their enhancements' constructor and the data it could hold.